src/idm/user.js

```diff
@@ -109,8 +109,56 @@ const idm = (base, token) => {
       })
       .then(res => (res))
       .catch(errorHandler);
+    },
+    /**
+    * @summary Reset password for any user. The user executing this action needs to be allowed to do this, e.g. admin.
+    * @name resetPassword
+    * @public
+    * @function
+    * @memberof agile.idm.user
+    * @param {String} userName user name
+    * @param {String} authType authentication type
+    * @param {String} newPassword new password
+    * @fulfil {Undefined}
+    * @returns {Promise}
+    * @example
+    * agile.idm.user.setPassword('bob','agile-local',"myNewPassword").then(function() {
+    *   console.log('password updated!');
+    * });
+    **/
+    resetPassword: (userName, authType, newPassword) => {
+      return instance.request({
+        method: 'PUT',
+        url: `${base}/api/v1/user/${userName}!@!${authType}/password`,
+        data: {'new_password': newPassword}
+      })
+      .then(res => (res))
+      .catch(errorHandler);
+    },
+    /**
+    * @summary update password for himself
+    * @name updatePassword
+    * @public
+    * @function
+    * @memberof agile.idm.user
+    * @param {String} oldPassword old password
+    * @param {String} newPassword new password
+    * @fulfil {Undefined}
+    * @returns {Promise}
+    * @example
+    * agile.idm.user.updatePassword("myOldPassword","myNewPassword").then(function() {
+    *   console.log('password updated!');
+    * });
+    **/
+    updatePassword: (oldPassword, newPassword) => {
+      return instance.request({
+        method: 'PUT',
+        url: `${base}/api/v1/user/password`,
+        data: {'old_password': oldPassword, 'new_password': newPassword}
```

Comment:
Nit: Object keys don't need strings here, axios serializes JavaScript objects to JSON for put/post.